### PR TITLE
Add release announcement for 3.12

### PR DESCRIPTION
**Pull request by melcorr**
There might be several errors in this. I ran towncrier this afternoon but maybe new features have been added since then. 
I have to admit defeat here. I have no idea what the words mean any more.

### CODE REVIEW

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  - announcement
+---
+
+Let's take a look at the main updates and enhancements as part of this release.
+
+This release also includes a plethora of bugfixes as well as some nice enhancements to Pulpcore and the Plugin API. For a full overview of changes, check out the [changlog](https://docs.pulpproject.org/pulpcore/changes.html).
+
+## Support added for automatic publishing and distributing [#7626](https://pulp.plan.io/issues/7626)
+
+When the content of a repository is updated, the repository can now be automatically published and redestributed.
```
Comment (mdellweg) [inline _posts/2021-04-08-pulp-3.12-is-generally-available.md]: Should we mention, that this work needs to be picked up by upcoming plugin releases to be used?

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+### Checking remote artifacts against ALLOWED_CONTENT_CHECKSUMS [#7985](https://pulp.plan.io/issues/7985)
+
+A new warning has been added to the on-demand sync process to check if remote artifacts are detected with checksums that do not correspond with the `ALLOWED_CONTENT_CHECKSUMS`.
+
+### New sync check for forbidden checksums [#8423](https://pulp.plan.io/issues/8423)
+
+During a sync of on-demand content, a new check is performed and throws an error if forbidden checksums are detected.
+
+As part of [#7989](https://pulp.plan.io/issues/7989) more enhancements updates were made to handle the exception that gets raised on the client side.
```
Comment (mdellweg) [inline _posts/2021-04-08-pulp-3.12-is-generally-available.md]: Sounds very similar to me. Can we collapse those two issues into one announcement?

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+### New logging options for ``DigestValidationError`` and ``SizeValidationError`` [#8357](https://pulp.plan.io/issues/8357)
+
+A `url` option has been added as a possible argument to ``DigestValidationError`` and ``SizeValidationError`` exceptions, so that you can better locate the url of the requested files.
+
+### A new Distribution MasterModel [#8384](https://pulp.plan.io/issues/8384)
+
+This release provides a new MasterModel `pulpcore.plugin.models.Distribution` that replaces the BaseDistribution `pulpcore.plugin.models.BaseDistribution` model.
+You will have to define a migration to move your data from `pulpcore.plugin.models.BaseDistribution` to `pulpcore.plugin.models.Distribution`. See the `pulp_file` migration 0009 as a reference example.[#8385](https://pulp.plan.io/issues/8385)
+
+This new model contains the `repository`, `repository_version`, and `publication` fields  instead of on the detail models as was done with `pulpcore.plugin.models.BaseDistribution`.
```
Comment (mdellweg) [inline _posts/2021-04-08-pulp-3.12-is-generally-available.md]:
```suggestion
This new model now contains the `repository`, `repository_version`, and `publication` fields that are supposed to move off of the detail models where they are located with `pulpcore.plugin.models.BaseDistribution`.
```

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+|BaseDistribution object|Replaced by|
+| --- | --- |
+|`pulpcore.plugin.models.BaseDistribution` |`pulpcore.plugin.models.Distribution` |
+|`pulpcore.plugin.viewset.BaseDistributionViewSet`|`pulpcore.plugin.viewset.DistributionViewSet` |
+|`pulpcore.plugin.serializer.BaseDistributionSerializer`|`pulpcore.plugin.serializer.DistributionSerializer`|
+|`pulpcore.plugin.serializer.PublicationDistributionSerializer`|`pulpcore.plugin.serializer.DistributionSerializer` |
+|`pulpcore.plugin.serializer.RepositoryVersionDistributionSerializer` | Define the `publication` field directly on your detail distribution object.|
+|`pulpcore.plugin.viewset.DistributionFilter` |`pulpcore.plugin.viewset.NewDistributionFilter`|
+|`pulpcore.plugin.tasking.enqueue_with_reservation` |`pulpcore.plugin.tasking.dispatch` |
```
Comment (mdellweg) [inline _posts/2021-04-08-pulp-3.12-is-generally-available.md]: This last deprecation is part of another change.
But i like the list being a table of deprecations for the entire release. So maybe adjust the headers instead.

Not sure how to add the "task arguments must be JSON serializable" deprecation into that table.

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 1 above)
Comment (melcorr) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: The fact that all the plugins have compatibility releases is a known fact, or you think I should be more specific? Doesn't this apply to all plugins and their ability to use enhancements in newer releases?

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 3 above)
Comment (melcorr) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: @mdellweg what do you mean by supposed to? Does it or doesn't it?

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 1 above)
Comment (mdellweg) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: To me the announcement here sounds like "You can use autopublish on _all_ your repos _now_."
But in fact, this lays the groundwork for plugins to support autopublish/-distribute.
So for expectation management, i would add something like:
"Look out for upcoming plugin releases to add this feature to their repositories."

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 4 above)
Comment (melcorr) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: Lol I will aim for that for lucky 3.13. I just removed this and wrote about it in the deprecations section. Thanks for catching this!

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 3 above)
Comment (melcorr) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: I added your sentence although I don't know what it means!

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 2 above)
Comment (melcorr) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: Removed the headings and just kept the initial headline.

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 1 above)
Comment (melcorr) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: OK this makes sense. Hope my rephrasing is adequate.

File: _posts/2021-04-08-pulp-3.12-is-generally-available.md (hunk 3 above)
Comment (mdellweg) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: The plugin migration you are mentioning before is supposed to move those fields.

Comment (melcorr) [reply _posts/2021-04-08-pulp-3.12-is-generally-available.md]: Ah I understand this now. Are we good to go with this section at least?

**Review by mdellweg — APPROVED** [review]

**Pull request merged**